ci-operator/step-registry/cucushift/installer/rehearse/gcp/cases/valid-confidential-computing/provision/cucushift-installer-rehearse-gcp-cases-valid-confidential-computing-provision-commands.sh

```diff
@@ -0,0 +1,256 @@
+  machineNetwork:
+  - cidr: 10.0.0.0/16
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  gcp:
+    projectID: ${GOOGLE_PROJECT_ID}
+    region: ${REGION}
+publish: External
+pullSecret: >
+  ${pull_secret}
+sshKey: |
+  ${ssh_pub_key}
+EOF
+}
+
+function compose_expected_err_msg()
+{
+  local -r instance_type="$1"
+  local -r compute_cc_type="$2"
+  local -r control_plane_cc_type="$3"
+  local -r on_host_maintenance="$4"
+
+  supported_cc_types=${instance_type_supported_cc_type_dict["${instance_type}"]}
+  unsupported_cc_types=${instance_type_unsupported_cc_type_dict["${instance_type}"]}
+
+  expected_err_msg=""
+  if [[ "${on_host_maintenance}" != "Terminate" ]]; then
+    expected_err_msg="controlPlane.platform.gcp.OnHostMaintenance: Invalid value: \\\"${on_host_maintenance}\\\": OnHostMaintenace must be set to Terminate when ConfidentialCompute is ${control_plane_cc_type}"
+  fi
+
+  set +e
+  if echo "${unsupported_cc_types}" | grep -qw "${control_plane_cc_type}"; then
+    if [[ "${expected_err_msg}" != "" ]]; then
+      expected_err_msg="${expected_err_msg}, "
+    fi
+    expected_err_msg="${expected_err_msg}controlPlane.platform.gcp.type: Invalid value: \\\"${instance_type}\\\": Machine type do not support ${control_plane_cc_type}. Machine types supporting ${control_plane_cc_type}: ${cc_type_machine_series_dict[${control_plane_cc_type}]}"
+  fi
+
+  if [[ "${on_host_maintenance}" != "Terminate" ]]; then
+    expected_err_msg="${expected_err_msg}, compute[0].platform.gcp.OnHostMaintenance: Invalid value: \\\"${on_host_maintenance}\\\": OnHostMaintenace must be set to Terminate when ConfidentialCompute is ${compute_cc_type}"
+  fi
+
+  if echo "${unsupported_cc_types}" | grep -qw "${compute_cc_type}"; then
+    if [[ "${expected_err_msg}" != "" ]]; then
+      expected_err_msg="${expected_err_msg}, "
+    fi
+    expected_err_msg="${expected_err_msg}compute[0].platform.gcp.type: Invalid value: \\\"${instance_type}\\\": Machine type do not support ${compute_cc_type}. Machine types supporting ${compute_cc_type}: ${cc_type_machine_series_dict[${compute_cc_type}]}"
+  fi
+  set -e
+}
+
+function random_choice()
+{
+  local -r cc_types_str="$1"
+
+  IFS=' ' read -r -a cc_types_array <<< "${cc_types_str}"
+  array_len="${#cc_types_array[@]}"
+  selected_index=$(( $RANDOM%($array_len) ))
+  selected_cc_type="${cc_types_array[${selected_index}]}"
+}
+
+function create_manifests()
+{
+  local -r instance_type="$1"
+  local -r cc_types_str="$2"
+  local -r on_host_maintenance="$3"
+  local -r testing_scenario_num="$4"
+  local -r success_flag="$5"
+
+  cluster_name="${CLUSTER_PREFIX}${testing_scenario_num}"
+  install_dir="/tmp/${cluster_name}"
+  mkdir -p "${install_dir}" 2>/dev/null
+  create_install_config "${cluster_name}" "${install_dir}"
+
+  random_choice "${cc_types_str}"
+  compute_cc_type="${selected_cc_type}"
+  random_choice "${cc_types_str}"
+  control_plane_cc_type="${selected_cc_type}"
+
+  cat > "${patch}" << EOF
+compute:
```

Comment:
Shall we also take `platform.gcp.defaultMachinePlatform.confidentialCompute/onHostMaintenance` into this testing?

Reply:
Got it, let me think about it, thanks!